### PR TITLE
added new email validator

### LINKED ISSUE #2560: Email validation not applying to invalid emails
**Describe the bug**
The email "e@il" is not a valid email address, but when offered as a value for email inputs it is valid and the forms can be submitted. 

**To Reproduce**
1. Go to any form with an email
2. Fill out the form to be successfull
3. Change the email address field to be "e@il"
4. See that you can submit the form

**Expected behavior**
Invalid email addresses should show errors and prevent form submission.

**Screenshots**
image

### PR DESCRIPTION
**Pull request by r-cochran**
### Issues Fixed
#2560

### CODE REVIEW

**Comment by MaximAndreevich** [comment]
Robert, we have additional requirement based on this validation. 
at the prompt-form-part.component.html 
For the primary-button. We want to pass .enabled() attribute to it. 
and be able to update it through the action. 
Are JMC team OK with this changes? Can we make it under this PR?

**Comment by r-cochran** [comment]
image
image
image

**Review by MaximAndreevich — DISMISSED** [review]

File: openpos-client-libs/projects/openpos-client-core-lib/src/lib/shared/validators/email-validator.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+import { ValidatorFn, FormControl, Validators } from '@angular/forms';
+
+export class EmailValidator implements IValidator {
+    name = this.constructor.name;
+
+    validationFunc: ValidatorFn = (ctrl: FormControl) => {
+        if (ctrl.value) {
+            const regex = new RegExp('^[a-z0-9._%+-]+@[a-z0-9.-]+\\.[a-z]{2,4}$');
+            const result = regex.test(ctrl.value);
+            console.log(`Regex is: ${regex}, control value is: ${ctrl.value}, result is: ${result}`);
```
Comment (jchirico8) [inline openpos-client-libs/projects/openpos-client-core-lib/src/lib/shared/validators/email-validator.ts:11]: Could you remove this log or add more context to it? I think it's too generic to be helpful

File: openpos-client-libs/projects/openpos-client-core-lib/src/lib/shared/validators/email-validator.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import { IValidator } from '../../core/interfaces/validator.interface';
+import { ValidatorFn, FormControl, Validators } from '@angular/forms';
+
+export class EmailValidator implements IValidator {
+    name = this.constructor.name;
+
+    validationFunc: ValidatorFn = (ctrl: FormControl) => {
+        if (ctrl.value) {
+            const regex = new RegExp('^[a-z0-9._%+-]+@[a-z0-9.-]+\\.[a-z]{2,4}$');
```
Comment (jchirico8) [inline openpos-client-libs/projects/openpos-client-core-lib/src/lib/shared/validators/email-validator.ts:9]: Are we supposed to force people to enter an all lowercase email?

Comment (r-cochran) [reply openpos-client-libs/projects/openpos-client-core-lib/src/lib/shared/validators/email-validator.ts:9]: good catch

**Review by chenson42 — APPROVED** [review]

**Pull request merged**